conda_recipe_manager/commands/rattler_bulk_build.py

```diff
@@ -0,0 +1,162 @@
+        {
+            "ignore_unknown_options": True,
+            "allow_extra_args": True,
+        },
+    ),
+)
+@click.argument("path", type=click.Path(exists=True, path_type=Path))  # type: ignore[misc]
+@click.option(
+    "--min-success-rate",
+    "-m",
+    type=click.FloatRange(0, 1),
+    default=DEFAULT_BULK_SUCCESS_PASS_THRESHOLD,
+    help="Sets a minimum passing success rate for bulk operations.",
+)
+@click.pass_context
+def rattler_bulk_build(ctx: click.Context, path: Path, min_success_rate: float) -> None:
+    """
+    Given a directory of feedstock repositories, performs multiple recipe builds using rattler-build.
+    All unknown options and arguments for this script are passed directly to `rattler-build build`.
+    NOTE:
+        - The build command is run as `rattler-build build -r <recipe.yaml> <ARGS>`
+        - rattler-build errors are dumped to STDERR
+    """
+    start_time: Final[float] = time.time()
+    files: Final[list[Path]] = []
+    for file_path in path.rglob(NEW_FORMAT_RECIPE_FILE_NAME):
+        files.append(file_path)
+
+    if not files:
+        print_err(f"No `recipe.yaml` files found in: {path}")
+        sys.exit(ExitCode.NO_FILES_FOUND)
+
+    # Process recipes in parallel
+    thread_pool_size: Final[int] = mp.cpu_count()
+    with mp.Pool(thread_pool_size) as pool:
+        results = dict(pool.starmap(build_recipe, [(file, path, ctx.args) for file in files]))  # type: ignore[misc]
+
+    # Gather statistics
+    total_recipes: Final[int] = len(files)
+    total_processed: Final[int] = len(results)
+    total_errors = 0
+    total_success = 0
+    recipes_with_errors: list[str] = []
+    error_histogram: dict[str, int] = {}
+    for file, build_result in results.items():
+        if build_result.code == ExitCode.SUCCESS:
+            total_success += 1
+        else:
+            total_errors += 1
+            recipes_with_errors.append(file)
+        if build_result.errors:
+            for error in build_result.errors:
+                if error not in error_histogram:
+                    error_histogram[error] = 0
+                error_histogram[error] += 1
+    percent_success: Final[float] = round(total_success / total_recipes, 2)
+
+    total_time: Final[float] = time.time() - start_time
+    stats = {
+        "total_recipe_files": total_recipes,
+        "total_recipes_processed": total_processed,
+        "total_errors": total_errors,
+        "percent_errors": round(total_errors / total_recipes, 2),
+        "percent_success": percent_success,
+        "timings": {
+            "total_exec_time": round(total_time, 2),
+            "avg_recipe_time": round(total_time / total_recipes, 2),
+            "thread_pool_size": thread_pool_size,
+        },
+    }
+    final_output = {
+        "recipes_with_build_error_code": recipes_with_errors,
+        "error_histogram": error_histogram,
+        "stats": stats,
+    }
+
+    print(json.dumps(final_output, indent=2))
```

Comment:
Do you want to print something to stderr in case we miss the success threshold?

Reply:
I've thought about that but I think I want to reserve STDERR for non-JSON logging. Since the CI pipeline will dump the JSON log with the percentage AND send a failed exit code, I feel like writing another message is a bit redundant.